RecoEgamma/EgammaElectronAlgos/src/GsfElectronAlgo.cc

```diff
@@ -562,8 +564,29 @@ reco::GsfElectronCollection GsfElectronAlgo::completeElectrons(edm::Event const&
     if (!electronData.calculateTSOS(mtsTransform, constraintAtVtx))
       continue;
 
-    createElectron(
-        electrons, electronData, eventData, caloTopology, caloGeometry, mtsTransform, magneticFieldInTesla, hoc);
+    eventData.retreiveOriginalTrackCollections(electronData.ctfTrackRef, electronData.coreRef->gsfTrack());
+
+    if (!eventData.originalCtfTracks.isValid()) {
+      eventData.originalCtfTracks = eventData.currentCtfTracks;
+    }
+
+    if (ctfTrackTable == std::nullopt) {
+      ctfTrackTable = edm::soa::makeTrackTable(*eventData.originalCtfTracks);
```

Comment:
The nice lazy table making mechanism https://github.com/cms-sw/cmssw/pull/30209 unfortunately can't be used here because the `originalCtfTracks` are not known in advance (meaning before starting the first electron loop).